src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+
+import laboneq._token
+import laboneq.simple as lo
+import numpy as np
+
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
+from qibolab.instruments.abstract import AbstractInstrument, InstrumentException
+from qibolab.paths import qibolab_folder
+from qibolab.pulses import FluxPulse, PulseSequence, PulseType
+from qibolab.sweeper import Parameter
+
+os.environ["LABONEQ_TOKEN"] = "ciao come va?"  # or other random text
+laboneq._token.is_valid_token = lambda _token: True
+
+# FIXME: Multiplex (For readout). Workaround integration weights padding with zeros.
+# FIXME: Handle on acquires for list of pulse sequences
+# FIXME: I think is a hardware limitation but I cant sweep multiple drive oscillator at the same time
+
+
+class ZhPulse:
+    """Zurich pulse from qibolab pulse"""
+
+    def __init__(self, pulse):
+        """Qibolab pulse"""
+        self.pulse = pulse
+        """Line associated with the pulse"""
+        self.signal = f"{pulse.type.name.lower()}{pulse.qubit}"
+        """Zurich pulse"""
+        self.zhpulse = self.select_pulse(pulse, pulse.type.name.lower())
+
+    # FIXME: Either implement more or create and arbitrary one
+    def select_pulse(self, pulse, pulse_type):
+        """Pulse translation"""
+        zh_pulse = None
+
+        if str(pulse.shape) == "Rectangular()":
+            zh_pulse = lo.pulse_library.const(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+            )
+        elif "Gaussian" in str(pulse.shape):
+            sigma = pulse.shape.rel_sigma
+            zh_pulse = lo.pulse_library.gaussian(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                zero_boundaries=False,
+            )
+
+        elif "GaussianSquare" in str(pulse.shape):
+            sigma = pulse.shape.rel_sigma
+            zh_pulse = lo.pulse_library.gaussian_square(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                width=round(pulse.duration * 1e-9, 9) * 0.9,  # 90% Flat
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                zero_boundaries=False,
+            )
+
+        elif "Drag" in str(pulse.shape):
+            sigma = pulse.shape.rel_sigma
+            beta = pulse.shape.beta
+            zh_pulse = lo.pulse_library.drag(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                beta=beta,
+                zero_boundaries=False,
+            )
+        elif "Slepian" in str(pulse.shape):
+            "Implement Slepian shaped flux pulse https://arxiv.org/pdf/0909.5368.pdf"
```

Comment:
Maybe keep this as a `# TODO`, rather than a string (I know you will need a `pass`, but that's a price we can pay)